### DISCUSSION TITLE
Multiple Scenes in a GeoDataset

### DISCUSSION THREAD

**Discussion by null-jones** [azavea/raster-vision#1735] [Q&A] [answered]
Howdy! I've been using the Raster Vision library for the last couple of weeks and I'm loving it so far.

Right now I'm trying to figure out how to use multiple scenes within a `SemanticSegmentationSlidingWindowGeoDataset`.  I can see this is pretty easy to do when [using Raster Vision as a pipeline](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/spacenet_vegas.py#L182), however I can't see an easy way to do it when using Raster Vision as a library.  I need to use multiple Scenes because some of my imagery is from a wide area with different labels, and Raster Vision tries to build a massive VRT which causes me to run out of memory pretty quickly.

Does anyone have any recommendations? Thanks in advance :)

**Comment by AdeelH** [comment, marked as answer]
Hi! You can create a `GeoDataset` for each scene and then concatenate them together using PyTorch's [`ConcatDataset`](https://pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset). This is in fact what Raster Vision [does internally](https://github.com/azavea/raster-vision/blob/0.20/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L1224-L1266) when using multiple scenes.

**Comment by null-jones** [comment]
Perfect, thank you so much for pointing me in the right direction Adeel!!
